tools/YamlSplitter/Program.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Tools.YamlSplitter
+{
+    using Microsoft.DocAsCode.Build.SchemaDriven;
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Tools.YamlSplitter.Models;
+
+    using CommandLine;
+    using Newtonsoft.Json.Schema;
+    using YamlDotNet.RepresentationModel;
+
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+
+    class Program
+    {
+        static HashSet<string> missingMergeKey = new HashSet<string>();
+
+        static void Main(string[] args)
+        {
+            CommandLine.Parser.Default.ParseArguments<InitOptions, UpdateOptions>(args)
+                .MapResult(
+                (InitOptions opts) => { Console.WriteLine("init mode is not supported right now."); return 0; },
+                (UpdateOptions opts) => RunUpdate(opts),
+                errs => -1);
+            foreach(var key in missingMergeKey)
+            {
+                Console.WriteLine(key + " does not have a merge key defined in schema, skipping...");
+            }
+        }
+
+        public static int RunUpdate(UpdateOptions opt)
+        {
+            opt.InputYamlFolder = Path.GetFullPath(opt.InputYamlFolder);
+            if (!string.IsNullOrEmpty(opt.OutputYamlFolder))
+            {
+                opt.OutputYamlFolder = Path.GetFullPath(opt.OutputYamlFolder);
+            }
+            else
+            {
+                opt.OutputYamlFolder = opt.InputYamlFolder;
+            }
+            if (!string.IsNullOrEmpty(opt.MDFolder))
+            {
+                opt.MDFolder = Path.GetFullPath(opt.MDFolder);
+            }
+            else
+            {
+                opt.MDFolder = opt.OutputYamlFolder;
+            }
+            var schemas = LoadSchemas(opt.SchemaFolder);
+            foreach (var ymlFile in Directory.EnumerateFiles(opt.InputYamlFolder, "*.yml", SearchOption.AllDirectories))
+            {
+                if (ymlFile.EndsWith("/toc.yml", StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+                var relativeYmlPath = PathUtility.MakeRelativePath(opt.InputYamlFolder, ymlFile);
+                var ymlOutputFile = Path.Combine(opt.OutputYamlFolder, relativeYmlPath);
+                var mdFile = Path.Combine(opt.MDFolder, relativeYmlPath + ".md");
+                ProcessFilePair(ymlFile, ymlOutputFile, mdFile, schemas);
+            }
+            return 0;
+        }
+
+        public static Dictionary<string, DocumentSchema> LoadSchemas(string schemaFolderPath)
```

Comment:
>public [](start = 8, length = 6)

does this function have been called in another place, why not make it "private", so as function `ProcessFilePair`